### PR TITLE
Fix: When workflow already executed. After controller restart, healthcheck could never be scheduled

### PR DESCRIPTION
**Pull request by carlyjiang**
Fix: When workflow already executed. After reconcile and there's no reschedule handler, the healthcheck will never be executed.

## Proposed Changes
  - When reconcile event, if it is executed and there's scheduled timer in controller handler, then directly return.
  - When scheduled timer executed, it should fetch latest healthcheck spec and submit workflow. (in case workflow has any change)
  
## Testing Done
  - [x] Add unit-tests
  - [x] Deploy healthcheck, repeatAfterSec changed from 900s to 180s, the frequency updated properly.
  - [x] Deploy healthcheck, repeatAfterSec changed from 90 to 180s, the frequency updated properly.
  - [x] Restarted controller, it will reconcile all healthchecks, and since there's no scheduled handler populated, workflow will be submitted and reschedule timer.

### CODE REVIEW

File: Dockerfile-local [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:latest
```
Comment (kevdowney) [inline Dockerfile-local:3]: Do we need another Dockerfile?

File: internal/controllers/healthcheck_controller.go [hunk 2]
```diff
@@ -421,18 +422,25 @@ func (r *HealthCheckReconciler) deleteRBACForWorkflow(ctx context.Context, log l
 // this function exists to assist with how a function called by the timer.AfterFunc() method operates to call a
 // function which takes parameters, it is easiest to build this closure which holds access to the parameters we need.
 // the helper returns a function object taking no parameters directly, this is what we want to give AfterFunc
-func (r *HealthCheckReconciler) createSubmitWorkflowHelper(ctx context.Context, log logr.Logger, wfNamespace string, hc *activemonitorv1alpha1.HealthCheck) func() {
+func (r *HealthCheckReconciler) createSubmitWorkflowHelper(ctx context.Context, log logr.Logger, wfNamespace string, prevHealthCheck *activemonitorv1alpha1.HealthCheck) func() {
 	return func() {
 		log.Info("Creating and Submitting Workflow...")
-		wfName, err := r.createSubmitWorkflow(ctx, log, hc)
+
+		healthCheckNew := &activemonitorv1alpha1.HealthCheck{}
```
Comment (kevdowney) [inline internal/controllers/healthcheck_controller.go:429]:
```suggestion
		healthCheckInstance := &activemonitorv1alpha1.HealthCheck{}
```
Change this to something more readable.

**Review by ZihanJiang96 — APPROVED** [review]

**Review by akhil-rane — APPROVED** [review]

File: Dockerfile-local (hunk 1 above)
Comment (carlyjiang) [reply Dockerfile-local:3]: This dockerfile is for local build testing purposes. mac m1 over Rancher had some problems to build amd64 image thru qemu, this is to move build stage to laptop and copy binary directly from laptop.

File: internal/controllers/healthcheck_controller.go (hunk 2 above)
Comment (carlyjiang) [reply internal/controllers/healthcheck_controller.go:429]: renamed the variables.

**Pull request merged**